src/api_recievers/influxqueries.js

```diff
@@ -0,0 +1,204 @@
+  return new Promise((resolve, reject) => {
+    const allTimes = []; // exact time of point
+    const allqueryIds = []; // Associated Query Session ID of that point
+    const allStartTimes = []; // When the measurement was started
+    const allEndTimes = []; // When the measreuemnt was ended 
+
+
+    // Get the current time
+    const timeOfCurrentSession = currentTime.seconds();
+    let queryClient = client.getQueryApi(org);
+    let stringVersion = `-${timeOfCurrentSession.toString()}s`;
+
+    // Construct the Flux query with the filter for the specific querySessionID
+    const fluxQuery = ` from(bucket: "WebsiteData") |> range(start: ${stringVersion}) |> filter(fn: (r) => r._measurement == "AppChange" and r.QuerySession == "${querySessionID}")
+    `;
+
+    queryClient.queryRows(fluxQuery, {
+      next: (row, tableMeta) => {
+        const tableObject = tableMeta.toObject(row);
+        allTimes.push(tableObject._time);
+        allqueryIds.push(tableObject.QuerySession);
+        allStartTimes.push(tableObject._start);
+        allEndTimes.push(tableObject._stop);
+        
+      },
+      error: (error) => {
+        reject(error);
+      },
+      complete: () => {
+        resolve({ allTimes, allqueryIds, allStartTimes, allEndTimes });
+      }
+    });
+  });
+};
+
+// Uses the query session and time gathered from the grab_time functions in order to 
+const SpecificStudySessionProcessing = async (querySessionid, _startTimeOfStudySession, _stopTimeOfStudySession) => {
+
+  try {
+    // Await the result from grab_times
+    const result = await grabTimesForStudySession(querySessionid);
+    let timesOfSessions = result.allTimes;
+    let idsOfSessions = result.allqueryIds;
+    let startTimes = result.allStartTimes;
+    let endTimes = result.allEndTimes;
+
+
+    //console.log('Times of Sessions:', times_of_sessions);
+    //console.log('Query IDs:', result.allqueryIds);
+    SpecificStudySession(timesOfSessions[0], timesOfSessions[(timesOfSessions.length) - 1], idsOfSessions[0], startTimes[0], endTimes[0]);
+
+  } catch (error) {
+    log.error('Error:', error);
+  }
+};
+
+const SpecificStudySession = (startTime, endTime, idsOfSessions, startTimes, endTimes) => {
+
+  timeOfCurrentSession = currentTime.seconds(); 
+
+  startDate = new Date(startTime);
+  endDate = new Date(endTime);
+  startTimes = new Date(startTimes);
+  endTimes = new Date(endTimes);
+  
+  log.debug("Start Time of Session: " + startDate);  
+  log.debug("End Time of Session:" + endDate);
+  log.debug("Session ID:" + " " + idsOfSessions);
+  log.debug("Start Time:" + " " + startTimes);
+  log.debug("End Time:" +" " + endTimes);
+
+
+  let queryClient = client.getQueryApi(org);
+
+  // let stringVersion = "-" + startTime;
+  //console.log(stringVersion);
+  
+
+  const fluxQuery = ` from(bucket: "WebsiteData") |> range(start: ${startTime}, stop: ${endTime}) |> filter(fn: (r) => r._measurement == "AppChange")`;
+
+  queryClient.queryRows(fluxQuery, {
```

Comment:
Why does this function not return any data after the query?

Reply:
It does I think? The function is commented out in this PR.